### PR TITLE
OvmfPkg/TdTcg2Dxe: Fix the SeparatorEvent issue in RTMRs

### PR DESCRIPTION
**Pull request by mxu9**
According to the TCG EFI platform specification, the firmware must measure the EV_SEPARATOR event into PCRs 0-7. Since PCRs 0-7 are mapped to RTMRs 0-1 and one EV_SEPARATOR event for one RTMRs 0-1 is expected.

An issue is found in TdTcg2Dxe that 2 EV_SEPARATOR events are measured to RTMR[0] but no EV_SEPARATOR event is measured to RTMR[1]. This patch fixes the above issue.

Cc: Erdem Aktas <erdemaktas@google.com>
Cc: Jiewen Yao <jiewen.yao@intel.com>
Cc: Gerd Hoffmann <kraxel@redhat.com>
Cc: Qinkun Bao <qinkun@google.com>
Cc: Tom Lendacky <thomas.lendacky@amd.com>
Cc: Michael Roth <michael.roth@amd.com>

# Description

<_Include a description of the change and why this change was made._>

<_For each item, place an "x" in between `[` and `]` if true. Example: `[x]` (you can also check items in GitHub UI)_>

<_Create the PR as a Draft PR if it is only created to run CI checks._>

<_Delete lines in \<\> tags before creating the PR._>

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

<_Describe the test(s) that were run to verify the changes._>

## Integration Instructions

<_Describe how these changes should be integrated. Use N/A if nothing is required._>

### CODE REVIEW

File: OvmfPkg/Tcg/TdTcg2Dxe/TdTcg2Dxe.c [hunk 1]
```diff
@@ -2160,11 +2160,17 @@ OnReadyToBoot (
...
     //
-    Status = MeasureSeparatorEvent (1);
+    // According to UEFI Spec 2.10 Section 38.4.1 the mapping between MrIndex and Intel
+    // TDX Measurement Register is:
+    //    MrIndex 0   <--> MRTD
+    //    MrIndex 1-3 <--> RTMR[0-2]
+    // RTMR[0] (i.e. MrIndex 1) is already done. So SepartorEvent shall be extended to
+    // RTMR[1] (i.e. MrIndex 2) as well.
+    //
+    Status = MeasureSeparatorEvent (2);
```
Comment (qinkunbao) [inline OvmfPkg/Tcg/TdTcg2Dxe/TdTcg2Dxe.c:2171]: nit: How about "Status = MeasureSeparatorEvent (CC_MR_INDEX_2_RTMR1);"?

Comment (mxu9) [reply OvmfPkg/Tcg/TdTcg2Dxe/TdTcg2Dxe.c:2171]: Good suggestion. Will update in next version.

**Review by qinkunbao — APPROVED** [review]
The description can be made more clear.

> Since PCRs 0-7 are mapped to RTMRs 0-1 and one EV_SEPARATOR event for one RTMRs 0-1 is expected.

As PCR[1] and PCR[7] map to RTMR[0], and PCRs [2-6] map to RTMR[1], it is necessary to measure one EV\_SEPARATOR event into RTMR[0] and another one into RTMR[1].

**Comment by mxu9** [comment]
> The description can be made more clear.
> 
> > Since PCRs 0-7 are mapped to RTMRs 0-1 and one EV_SEPARATOR event for one RTMRs 0-1 is expected.
> 
> As PCR[1] and PCR[7] map to RTMR[0], and PCRs [2-6] map to RTMR[1], it is necessary to measure one EV_SEPARATOR event into RTMR[0] and another one into RTMR[1].

Commit message is updated.

**Review by jyao1 — APPROVED** [review]

**Comment by qinkunbao** [comment]
Hi @mxu9 and @jyao1 , we've tested the fix in the TDX machine and it appears to be working well. When can we expect the PR to be merged?

**Comment by jyao1** [comment]
I think it can be merged after code freeze.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**